### PR TITLE
[BUG][BE-12113]fix htrace dependency conflicts

### LINKED ISSUE #12113: [Bug] [HDFS] enable HDFS cannot find class htrace-core
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

enable HDFS will throw ClassNotFoundException: htrace-core
version 3.1.0
c9b353dd8657fd9940fe30127cac1bf

6a3b56af9f18d7ea97537cc3cd3b75f

### What you expected to happen

HDFS works fine

### How to reproduce

change HDFS settings, then will throw exception

### Anything else

will also check dev whether have this problem

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request
close https://github.com/apache/dolphinscheduler/issues/12113
fix htrace dependency conflicts
8a7bc93cdba0ee455fad3de0dc45932

## Brief change log

fix htrace dependency conflicts

## Verify this pull request

Manually verified the change by testing locally.
If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
@Tianqi-Dotes please check the dependency license.

File: dolphinscheduler-api/pom.xml [hunk 1]
```diff
@@ -190,11 +190,23 @@
             <artifactId>spring-cloud-starter-kubernetes-fabric8-config</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml:193]: Please replace this with 

```xml
             <dependency>
                 <groupId>org.apache.hbase.thirdparty</groupId>
                 <artifactId>hbase-noop-htrace</artifactId>
             </dependency>
```

File: dolphinscheduler-api/pom.xml [hunk 2]
```diff
@@ -190,11 +190,23 @@
...
+            <version>${htrace.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+
         <dependency>
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml]: This should be removed.

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]
Please replace all htrace-core with hbase-noop-htrace. We don't need tracing and the htrace has CVEs.

File: pom.xml [hunk 3]
```diff
@@ -92,6 +92,7 @@
         <docker.tag>${project.version}</docker.tag>
         <docker.build.skip>true</docker.build.skip>
         <docker.push.skip>true</docker.push.skip>
+        <htrace.version>4.2.0-incubating</htrace.version>
```
Comment (caishunfeng) [inline pom.xml]: Should manage in bom.

**Comment by caishunfeng** [comment]
related pr: #11881

File: dolphinscheduler-api/pom.xml [hunk 4]
```diff
@@ -190,6 +190,13 @@
             <artifactId>spring-cloud-starter-kubernetes-fabric8-config</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
+            <scope>runtime</scope>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml]: These are unnecessary. Please remove

```suggestion
```

File: dolphinscheduler-common/pom.xml [hunk 5]
```diff
@@ -188,6 +188,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
```
Comment (kezhenxu94) [inline dolphinscheduler-common/pom.xml]: Same here

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml [hunk 6]
```diff
@@ -78,6 +78,12 @@
             <artifactId>spring-jdbc</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
```
Comment (kezhenxu94) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml]: Here

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml [hunk 7]
```diff
@@ -127,6 +133,10 @@
                     <artifactId>log4j</artifactId>
                     <groupId>log4j</groupId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.apache.htrace</groupId>
+                    <artifactId>htrace-core4</artifactId>
+                </exclusion>
```
Comment (kezhenxu94) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml]: These are unnecessary

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml [hunk 8]
```diff
@@ -42,6 +42,12 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
```
Comment (kezhenxu94) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml]: Remove

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml [hunk 9]
```diff
@@ -91,6 +97,10 @@
                     <artifactId>log4j</artifactId>
                     <groupId>log4j</groupId>
                 </exclusion>
+                <exclusion>
```
Comment (kezhenxu94) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml]: Remove

File: dolphinscheduler-master/pom.xml [hunk 10]
```diff
@@ -95,6 +95,12 @@
             <artifactId>caffeine</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
```
Comment (kezhenxu94) [inline dolphinscheduler-master/pom.xml]: Remove

File: dolphinscheduler-master/pom.xml [hunk 11]
```diff
@@ -148,6 +154,10 @@
                     <groupId>org.slf4j</groupId>
                     <artifactId>slf4j-reload4j</artifactId>
                 </exclusion>
+                <exclusion>
```
Comment (kezhenxu94) [inline dolphinscheduler-master/pom.xml]: Remove

File: dolphinscheduler-server/pom.xml [hunk 12]
```diff
@@ -140,6 +140,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hbase.thirdparty</groupId>
+            <artifactId>hbase-noop-htrace</artifactId>
+            <version>${htrace.version}</version>
```
Comment (kezhenxu94) [inline dolphinscheduler-server/pom.xml]: Remove

File: pom.xml [hunk 13]
```diff
@@ -92,6 +92,7 @@
         <docker.tag>${project.version}</docker.tag>
         <docker.build.skip>true</docker.build.skip>
         <docker.push.skip>true</docker.push.skip>
+        <htrace.version>4.1.1</htrace.version>
```
Comment (kezhenxu94) [inline pom.xml]: Remove this

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/pom.xml (hunk 1 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-api/pom.xml:193]: done

File: dolphinscheduler-api/pom.xml (hunk 2 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-api/pom.xml]: done

File: pom.xml (hunk 3 above)
Comment (Tianqi-Dotes) [reply pom.xml]: done

File: dolphinscheduler-api/pom.xml (hunk 4 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-api/pom.xml]: done

File: dolphinscheduler-common/pom.xml (hunk 5 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-common/pom.xml]: done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml (hunk 6 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml]: done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml (hunk 7 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/pom.xml]: done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml (hunk 8 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml]: done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml (hunk 9 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hive/pom.xml]: done

File: dolphinscheduler-master/pom.xml (hunk 10 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-master/pom.xml]: done

File: dolphinscheduler-server/pom.xml (hunk 12 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-server/pom.xml]: done

File: pom.xml (hunk 13 above)
Comment (Tianqi-Dotes) [reply pom.xml]: done

**Review by kezhenxu94 — APPROVED** [review]

**Comment by caishunfeng** [comment]
Thanks~

**Pull request merged**